pkg/anonymization/anonymizer.go

```diff
@@ -0,0 +1,162 @@
+	}, nil
+}
+
+// NewAnonymizer creates a new instance of anonymizer with a provided config observer and openshift config client
+func NewAnonymizerFromConfigClient(
+	ctx context.Context, configObserver *configobserver.Controller, configClient configv1client.ConfigV1Interface,
+) (*Anonymizer, error) {
+	baseDomain, err := utils.GetClusterBaseDomain(ctx, configClient)
+	if err != nil {
+		return nil, err
+	}
+
+	networksConfig, err := configClient.Networks().Get(ctx, "cluster", metav1.GetOptions{})
+	if err != nil {
+		return nil, err
+	}
+
+	var networks []string
+	for _, network := range networksConfig.Spec.ClusterNetwork {
+		networks = append(networks, network.CIDR)
+	}
+	for _, network := range networksConfig.Spec.ServiceNetwork {
+		networks = append(networks, network)
+	}
+	for _, network := range networksConfig.Spec.ExternalIP.AutoAssignCIDRs {
+		networks = append(networks, network)
+	}
+	for _, network := range networksConfig.Spec.ExternalIP.Policy.AllowedCIDRs {
+		networks = append(networks, network)
+	}
+	for _, network := range networksConfig.Spec.ExternalIP.Policy.RejectedCIDRs {
+		networks = append(networks, network)
+	}
+
+	return NewAnonymizer(configObserver, baseDomain, networks)
+}
+
+// AnonymizeMemoryRecord takes record.MemoryRecord, removes the sensitive data from it and returns the same object
+func (anonymizer *Anonymizer) AnonymizeMemoryRecord(memoryRecord *record.MemoryRecord) *record.MemoryRecord {
+	if anonymizer.configObserver == nil {
+		return memoryRecord
+	}
+
+	if !anonymizer.configObserver.Config().EnableGlobalObfuscation {
+		return memoryRecord
+	}
+
+	if len(anonymizer.clusterBaseDomain) != 0 {
+		memoryRecord.Data = bytes.ReplaceAll(
+			memoryRecord.Data,
+			[]byte(anonymizer.clusterBaseDomain),
+			[]byte(ClusterBaseDomainPlaceholder),
+		)
+		memoryRecord.Name = strings.ReplaceAll(
+			memoryRecord.Name,
+			anonymizer.clusterBaseDomain,
+			ClusterBaseDomainPlaceholder,
+		)
+	}
+
+	// We could use something like https://github.com/yl2chen/cidranger but we shouldn't typically have many networks
+	// so it's fine to just iterate over them
+
+	memoryRecord.Data = anonymizer.ipRegex.ReplaceAllFunc(memoryRecord.Data, func(originalIPBytes []byte) []byte {
+		originalIP := net.ParseIP(string(originalIPBytes))
+		if originalIP == nil {
+			klog.Warningf("Unable to parse IP '%v'", string(originalIPBytes))
+			// Unable to parse an IP, so just return whatever it is. It shouldn't happen.
+			return originalIPBytes
+		}
+
+		isIPv4 := originalIP.To4() != nil
+
+		if !isIPv4 {
+			// TODO: to be implemented later
+			// the problem is that some strings can be incorrectly identified as ip v6
+			// we can try looking only for those which are wrapped by quotes or something
+			return originalIPBytes
+		}
+
+		for _, network := range anonymizer.networks {
+			if network.Contains(originalIP) {
+				// return the first matched network
+				return []byte(network.IP.String())
+			}
```

Comment:
I guess you know it, but this won't match all the IPs in the archive  - e.g `hostIP` in `HostSubnet` definition and more...

Reply:
Well, it will become 0.0.0.0